src/socket-proxy/socket-proxyd.c

```diff
@@ -533,23 +562,42 @@ static int add_listen_socket(Context *context, int fd) {
         return 0;
 }
 
+static int idle_time_cb(sd_event_source *s, uint64_t usec, void *userdata) {
+        Context *c = userdata;
+        int r;
+
```

Comment:
Please add a check here to immediately exit without doing anything (except maybe a warning message)when `!set_isempty(context->connections)`. (The return value from sd_event_source_set_enabled() is ignored above, so even though this should never ever fail, let's be consistent, and assume here that the call might have failed.)

Reply:
It now returns early (to become a noop), with warning message.